app/Http/Controllers/Campaign/GalleryController.php

```diff
@@ -125,6 +126,34 @@ public function edit(Image $image)
         return view('gallery.edit', compact('image', 'folders'));
     }
 
+    /**
+     * @param StoreImageFocus $request
+     * @param Entity $entity
+     * @return \Illuminate\Http\RedirectResponse
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function saveFocus(StoreImageFocus $request, Image $image)
+    {
+        $campaign = CampaignLocalization::getCampaign();
+        $this->authorize('gallery', $campaign);
+
+        $image->focus_x = $request->post('focus_x');
```

Comment:
Please move the code into the service instead of the controller so that we can re-use the code elsewhere if needed :-)